File: 7kyu 1/DNAStrand.js
function DNAStrand(dna) {
   let res = ""
   for (let i = 0; i < dna.length; i++) {
      if (dna[i] == "A") {
         res += "T"
      } else if (dna[i] == "T") {
         res += "A"
      } else if (dna[i] == "G") {
         res += "C"
      } else {
         res += "G"
      }
   }
   return res;
}


const DNAStrand = dna => dna.replace(/./g, m => 'CGAT' ['GCTA'.indexOf(m)]);


function DNAStrand(dna) {
   let table = {
      C: 'G',
      G: 'C',
      A: 'T',
      T: 'A'
   };
   return dna.split('').map(function (cv) {
      return table[cv];
   }).join('');
}


function DNAStrand(dna) {
   return dna.replace(/A/g, 't').replace(/T/g, 'a').replace(/C/g, 'g').replace(/G/g, 'c').toUpperCase();
}


function DNAStrand(dna) {
   return dna.split('').map(function (v) {
      return {
         A: 'T',
         T: 'A',
         C: 'G',
         G: 'C'
      } [v];
   }).join('');
}


console.log(DNAStrand("AAAA"), "TTTT", "String AAAA is")
console.log(DNAStrand("ATTGC"), "TAACG", "String ATTGC is")
console.log(DNAStrand("GTAT"), "CATA", "String GTAT is")

File: Python/7py 1/DNA_strand.py
"""
|--------------------------------------------------------------------------
| Complementary DNA
|--------------------------------------------------------------------------
|
"""

def DNA_strand(dna):
   res = ""
   for i in range(len(dna)):
      res += 'T' if dna[i] == 'A' else 'A' if dna[i] == 'T' else 'C' if dna[i] == 'G' else 'G'
   return res


import string
def DNA_strand(dna):
   return dna.translate(string.maketrans("ATCG","TAGC"))


def DNA_strand(dna):
   reference = { "A":"T", "T":"A", "C":"G", "G":"C" }
   return "".join([reference[x] for x in dna])


def DNA_strand(dna):
   return ''.join([{'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G'}[letter] for letter in dna])


print(DNA_strand("AAAA"),"TTTT","String AAAA is")
print(DNA_strand("ATTGC"),"TAACG","String ATTGC is")
print(DNA_strand("GTAT"),"CATA","String GTAT is")